### PR TITLE
Dep bump 2022-04-22

### PR DESCRIPTION
**Pull request by Jimbo4350**

### CODE REVIEW

**Review by coot — APPROVED** [review]
LGTM

**Review by amesgen — CHANGES_REQUESTED** [review]
I ported the tracing logic for the pipelining events (from https://github.com/input-output-hk/cardano-node/pull/3764) in [this patch](https://gist.github.com/amesgen/cecfbc3a558aacae7ec672def880cc7d). This also includes a change to `doelide`; there will be a follow-up PR at some point (but this is not blocking this PR).

Apart from that, LGTM.

**Comment by Jimbo4350** [comment]
> there will be a follow-up PR at some point (but this is not blocking this PR).

@amesgen  if this is not blocking the PR, why are you requesting changes?

**Comment by amesgen** [comment]
I was referring to the follow-up PR, it will refine the `doelide` logic. The patch I linked to should be applied in this PR.

**Review by amesgen — APPROVED** [review]

**Comment by jbgi** [comment]
Fix for the cardano-node-edge test is at https://github.com/input-output-hk/ouroboros-network/pull/3729

**Review by newhoggy — APPROVED** [review]

**Comment by newhoggy** [comment]
bors merge

**Comment by newhoggy** [comment]
I backed out the last two commits as per @Jimbo4350's instruction and will try merging again to see if that helped.

**Comment by Jimbo4350** [comment]
@newhoggy I think https://github.com/input-output-hk/ouroboros-network/pull/3729 has to be merged first. My commits don't fix the problem.

**Review by MarcFontaine — APPROVED** [review]

File: cardano-api/src/Cardano/Api/Block.hs [hunk 1]
```diff
@@ -165,7 +170,7 @@ getBlockTxs (ByronBlock Consensus.ByronBlock { Consensus.byronBlockRaw }) =
         } -> map ByronTx txs
 getBlockTxs (ShelleyBlock era Consensus.ShelleyBlock{Consensus.shelleyBlockRaw}) =
     obtainConsensusShelleyBasedEra era $
-      getShelleyBlockTxs era shelleyBlockRaw
+      getShelleyBlockTxs era shelleyBlockRaw -- HERE
```
Comment (newhoggy) [inline cardano-api/src/Cardano/Api/Block.hs:173]:
```
src/Cardano/Api/Block.hs:173:30: error:
    • Could not deduce: Ledger.Crypto (ShelleyLedgerEra era)
                        ~ Consensus.StandardCrypto
      from the context: Consensus.ShelleyBasedEra (ShelleyLedgerEra era)
        bound by a type expected by the context:
                   Consensus.ShelleyBasedEra (ShelleyLedgerEra era) => [Tx era]
        at src/Cardano/Api/Block.hs:173:7-44
      Expected type: Ledger.Block
                       (TPraos.BHeader (Ledger.Crypto (ShelleyLedgerEra era)))
                       (ShelleyLedgerEra era)
        Actual type: Ledger.Block
                       (Ouroboros.Consensus.Shelley.Protocol.Abstract.ShelleyProtocolHeader
                          (Consensus.TPraos Consensus.StandardCrypto))
                       (ShelleyLedgerEra era)
    • In the second argument of 'getShelleyBlockTxs', namely
        'shelleyBlockRaw'
      In the second argument of '($)', namely
        'getShelleyBlockTxs era shelleyBlockRaw'
      In the expression:
        obtainConsensusShelleyBasedEra era
          $ getShelleyBlockTxs era shelleyBlockRaw
    • Relevant bindings include
        shelleyBlockRaw :: Ledger.Block
                             (Ouroboros.Consensus.Shelley.Protocol.Abstract.ShelleyProtocolHeader
                                (Consensus.TPraos Consensus.StandardCrypto))
                             (ShelleyLedgerEra era)
          (bound at src/Cardano/Api/Block.hs:171:54)
        era :: ShelleyBasedEra era
          (bound at src/Cardano/Api/Block.hs:171:27)
        getBlockTxs :: Block era -> [Tx era]
          (bound at src/Cardano/Api/Block.hs:162:1)
    |
173 |       getShelleyBlockTxs era shelleyBlockRaw
    |                              ^^^^^^^^^^^^^^^
```

File: cardano-api/src/Cardano/Api/Block.hs [hunk 2]
```diff
@@ -299,7 +304,7 @@ getBlockHeader (ShelleyBlock shelleyEra block) = case shelleyEra of
       where
         Consensus.HeaderFields {
             Consensus.headerFieldHash
-              = Consensus.ShelleyHash (TPraos.HashHeader (Cardano.Crypto.Hash.Class.UnsafeHash hashSBS)),
+              = Consensus.ShelleyHash (TPraos.HashHeader (Cardano.Crypto.Hash.Class.UnsafeHash hashSBS)), -- HERE
```
Comment (newhoggy) [inline cardano-api/src/Cardano/Api/Block.hs:307]:
```
src/Cardano/Api/Block.hs:173:30: error:
    • Could not deduce: Ledger.Crypto (ShelleyLedgerEra era)
                        ~ Consensus.StandardCrypto
      from the context: Consensus.ShelleyBasedEra (ShelleyLedgerEra era)
        bound by a type expected by the context:
                   Consensus.ShelleyBasedEra (ShelleyLedgerEra era) => [Tx era]
        at src/Cardano/Api/Block.hs:173:7-44
      Expected type: Ledger.Block
                       (TPraos.BHeader (Ledger.Crypto (ShelleyLedgerEra era)))
                       (ShelleyLedgerEra era)
        Actual type: Ledger.Block
                       (Ouroboros.Consensus.Shelley.Protocol.Abstract.ShelleyProtocolHeader
                          (Consensus.TPraos Consensus.StandardCrypto))
                       (ShelleyLedgerEra era)
    • In the second argument of 'getShelleyBlockTxs', namely
        'shelleyBlockRaw'
      In the second argument of '($)', namely
        'getShelleyBlockTxs era shelleyBlockRaw'
      In the expression:
        obtainConsensusShelleyBasedEra era
          $ getShelleyBlockTxs era shelleyBlockRaw
    • Relevant bindings include
        shelleyBlockRaw :: Ledger.Block
                             (Ouroboros.Consensus.Shelley.Protocol.Abstract.ShelleyProtocolHeader
                                (Consensus.TPraos Consensus.StandardCrypto))
                             (ShelleyLedgerEra era)
          (bound at src/Cardano/Api/Block.hs:171:54)
        era :: ShelleyBasedEra era
          (bound at src/Cardano/Api/Block.hs:171:27)
        getBlockTxs :: Block era -> [Tx era]
          (bound at src/Cardano/Api/Block.hs:162:1)
    |
173 |       getShelleyBlockTxs era shelleyBlockRaw
    |                              ^^^^^^^^^^^^^^^

src/Cardano/Api/Block.hs:307:40: error:
    • Couldn't match type 'TPraos.HashHeader crypto0'
                     with 'Cardano.Crypto.Hash.Class.Hash
                             (Cardano.Ledger.Crypto.HASH Consensus.StandardCrypto)
                             Cardano.Ledger.Hashes.EraIndependentBlockHeader'
      Expected type: Cardano.Ledger.Keys.Hash
                       Consensus.StandardCrypto
                       Cardano.Ledger.Hashes.EraIndependentBlockHeader
        Actual type: TPraos.HashHeader crypto0
    • In the pattern:
        TPraos.HashHeader (Cardano.Crypto.Hash.Class.UnsafeHash hashSBS)
      In the pattern:
        Consensus.ShelleyHash (TPraos.HashHeader (Cardano.Crypto.Hash.Class.UnsafeHash hashSBS))
      In the pattern:
        Consensus.HeaderFields {Consensus.headerFieldHash = Consensus.ShelleyHash (TPraos.HashHeader (Cardano.Crypto.Hash.Class.UnsafeHash hashSBS)),
                                Consensus.headerFieldSlot, Consensus.headerFieldBlockNo}
    |
307 |               = Consensus.ShelleyHash (TPraos.HashHeader (Cardano.Crypto.Hash.Class.UnsafeHash hashSBS)),
    |                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by nc6** [comment]
Closing in favour of #3818

**Comment by Jimbo4350** [comment]
It is easier to separate concerns if this is merged prior to #3818

**Pull request closed without merging**